components/CreateForm.js

```diff
@@ -1,57 +1,142 @@
 import { useState, useRef } from "react";
 import styled from "styled-components";
 import { Button, ButtonContainer } from "./Buttons";
+import { TbCheck, TbPlus } from "react-icons/tb";
+import { nanoid, customAlphabet } from "nanoid";
+
+const slugSuffix = customAlphabet(
+  "23456789abcdefghklmnpqrstuvwxyzABCDEFGHKLMNPQRSTUVWXYZ",
+  4
+);
 
 function CreateForm() {
-  const [steps, setSteps] = useState(1);
   const buttonRef = useRef();
+  const [inputSteps, setInputSteps] = useState([
+    { step: 1, stepTitle: "", stepUrl: "", stepDescription: "" },
+  ]);
+  const [inputTutorialTitle, setInputTutorialTitle] = useState("");
 
-  function scrollToButton() {
-    setTimeout(() => {
-      buttonRef.current.scrollIntoView({ behavior: "smooth" });
-    });
+  function handleTitleChange(event) {
+    let titleInput = event.target.value;
+    // Thank you, https://github.com/Roland-Hufnagel
+    titleInput = titleInput.startsWith(" ")
+      ? titleInput.replace(" ", "")
+      : titleInput.includes("  ")
+      ? titleInput.replace("  ", " ")
+      : titleInput;
+    setInputTutorialTitle(titleInput);
+  }
+
+  function handleFormChange(index, event) {
+    const data = [...inputSteps];
+    if (event.target.value.trim() === "") {
+      data[index][event.target.name] = "";
+    } else if (event.target.value[0] === " ") {
+      data[index][event.target.name] = event.target.value.trim();
+    } else {
+      data[index][event.target.name] = event.target.value;
+    }
+    setInputSteps(data);
   }
 
   function handleAddStep() {
-    setSteps((prevSteps) => prevSteps + 1);
+    const additionalStep = {
+      step: inputSteps.length + 1,
+      stepTitle: "",
+      stepUrl: "",
+      stepDescription: "",
+    };
+    setInputSteps((prevInputSteps) => [...prevInputSteps, additionalStep]);
     scrollToButton();
   }
 
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (inputTutorialTitle.replace(/[^a-zA-Z0-9]/g, "").length < 5) {
+      alert(
+        "The tutorial title must not consist of less than five letters and numbers (a-Z, 0-9, no special characters)."
+      );
+      return;
+    }
+    const newTutorial = {
+      id: nanoid(),
+      name: inputTutorialTitle,
+      cover: inputSteps[inputSteps.length - 1]["stepUrl"],
+      slug: inputTutorialTitle
+        .toLowerCase()
+        .replace(/ /g, "-")
+        .replace(/[-]+/g, "-")
+        .replace(/[^\w-]+/g, "")
+        .concat("-", slugSuffix()),
+      steps: [...inputSteps],
```

Comment:
you don't need to make a copy here from the inputsteps:
```suggestion
      steps: inputSteps,
```